sycl/include/sycl/ext/oneapi/experimental/annotated_usm/alloc_util.hpp

```diff
@@ -91,13 +91,12 @@ struct ValidAllocPropertyList<T, detail::properties_t<Prop, Props...>>
                  IsRuntimePropertyValid<Prop>::value) &&
                     ValidAllocPropertyList<
                         T, detail::properties_t<Props...>>::value> {
+  static_assert(is_property_value_v<Prop>);
   // check if a compile-time property is valid for annotated_ptr
-  static_assert(!detail::IsCompileTimePropertyValue<Prop>::value ||
-                    is_valid_property<T *, Prop>::value,
+  static_assert(!std::is_empty_v<Prop> || is_valid_property<T *, Prop>::value,
```

Comment:
Could we consider aliasing `std::is_empty_v` to make the use of it here clearer? It might not be clear that it has to do with whether or not the property has runtime contents.

Reply:
Sure, but ideally we need to refactor it one level higher - this entire trait needs to lifted to C++17 at the very least (with pack expansion).